### ISSUE TITLE
App crashing on release (android)

### ISSUE THREAD

**Issue by stavgafny** [Dev-hwang/flutter_foreground_task#106]
So I wasn't able to make the app work on release.
Compiling the example app on release raised no issues but trying to launch the app causes it to crash with:
```
E/flutter (14495): [ERROR:flutter/runtime/dart_isolate.cc(707)] Could not resolve main entrypoint function.
E/flutter (14495): [ERROR:flutter/runtime/dart_isolate.cc(191)] Could not run the run main Dart entrypoint.
E/flutter (14495): [ERROR:flutter/runtime/runtime_controller.cc(382)] Could not create root isolate.
E/flutter (14495): [ERROR:flutter/shell/common/shell.cc(574)] Could not launch engine with configuration.
```
Eventually I found out what the problem was, after a lot of digging around and searching with not much results or information about it I decided to post here if anyone else is tackling the same issue.

Basically on android you have the `activity` that is responsible for the main UI thread(your main code), by default android doesn't allow running processes of closed apps since android 8 to prevent slowing down and draining the battery, so for an app to have a process running even if closed they introduced `service` which enables you to do that with an indicating notification as long as the process is running.
This is why you need to put the \<service\> tag on the **AndroidManifest.xml** file for the foreground task to work at all, so the problem on release is that the app doesn't know from where to start the foreground task isolate (entry point).
On the example code I noticed that the function `startCallback` has to be on the global scope and trying to make it an anonymous function won't work even on debug, this means that that's the main entry point of the service so all I had to do was add `@pragma('vm:entry-point')`
above the function and it fixed the issue and worked smoothly as expected on release.

### How the solution looks:
```
@pragma('vm:entry-point')
void startCallback() {
  FlutterForegroundTask.setTaskHandler(FirstTaskHandler());
}
```
Don't know if it works on ios but I do know that it also happens there so it might be from the same reason.

**Comment by GaelleJoubert** [comment]
I had the exact same issue in release.
 I try to add the `@pragma('vm:entry-point')` , but now my app is actually crashing instead of displaying the `Dart Error: Dart_LookupLibrary ...`.
 It is working fine in debug.

**Comment by GaelleJoubert** [comment]
I have explored a little bit more about this crashing app and found this cause of the issue and a possible workaround.
This is all detailed in a new issue:  https://github.com/Dev-hwang/flutter_foreground_task/issues/116

**Comment by Faiizii** [comment]
my app is also crashing in release mode only after starting the service.
